src/ACEtk/block/details/BaseDistributionData/src/ctor.hpp

```diff
@@ -0,0 +1,48 @@
+BaseDistributionData() = default;
+
+BaseDistributionData( const BaseDistributionData& ) = default;
+BaseDistributionData( BaseDistributionData&& ) = default;
+
+/**
+ *  @brief Constructor
+ *
+ *  @param[in] name             the name of the block
+ *  @param[in] boundaries       the interpolation range boundaries
+ *  @param[in] interpolants     the interpolation types for each range
+ *  @param[in] distributions    the distributions for each incident energy
+ *  @param[in] locb             the starting xss index with respect to the superblock
+ */
+BaseDistributionData( std::string&& name,
+                      std::vector< long >&& boundaries,
+                      std::vector< long >&& interpolants,
+                      std::vector< Distribution >&& distributions,
+                      std::size_t locb = 1 ) :
+  Base( std::move( name ),
+        generateXSS( std::string( name ),
+                     std::move( boundaries ), std::move( interpolants ),
+                     std::move( distributions ), locb ) ),
+  locb_( locb ) {
+
+  this->generateBlocks();
+}
+
+/**
+ *  @brief Constructor
+ *
+ *  @param[in] name             the name of the block
+ *  @param[in] locb    the starting xss index with respect to the superblock
+ *  @param[in] sig     the begin iterator of the block in the XSS array
+ *  @param[in] end     the end iterator of the block in the XSS array
+ */
+BaseDistributionData( std::string&& name, std::size_t locb,
+                      Iterator begin, Iterator end ) :
+  Base( std::move( name ), begin, end ), locb_( locb ) {
+
+  std::size_t nr = static_cast< std::size_t >( this->XSS( 1 ) );
```

Comment:
[clang-tidy, recommended]
`nr` and `ne` can be marked const. Also, since the type is declared in the `static_cast`, suggest just making these `const auto`.

Reply:
I'm not being consistent. I actually do this in other locations.